qiskit/transpiler/target.py

```diff
@@ -1045,6 +1048,10 @@ def build_coupling_map(self, two_q_gate=None):
         else:
             return None
 
+    def add_measuregrouping(self, meas_map=None):
```

Comment:
You can update the constructor of `Target` with meas_map (because this is hardware constraints and user cannot modify once it's set). Then store it as a protected member and provide only property method for it so that user cannot override.